### PR TITLE
Run Horovod on Ray

### PR DESCRIPTION
**Pull request by yangw1234**
## New Feature
Create a `HorovodRayRunner` that can run user horovod func on ray.

## Usage
```python
def horovod_func():
   hvd.init()
   ...
   model.train()

runner = HorovodRayTrainer(ray_context)
runner.run(horovod_func)
```

## Testing

- [x] local manually started ray
- [x] local RayOnSpark
- [x] yarn cluster
- [x] first need to upgrade ray to 0.8.4 pending #2249

### CODE REVIEW

**Comment by yangw1234** [comment]
jenkins pass http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/3332/

File: pyzoo/zoo/orca/learn/horovod/horovod_ray_trainer.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+            key = (alloc_info.hostname, alloc_info.local_rank)
+            local_envs = self.per_worker_envs[name_rank_to_id[key]]
+            local_envs["HOROVOD_RANK"] = str(alloc_info.rank)
+            local_envs["HOROVOD_SIZE"] = str(alloc_info.size)
+            local_envs["HOROVOD_LOCAL_RANK"] = str(alloc_info.local_rank)
+            local_envs["HOROVOD_LOCAL_SIZE"] = str(alloc_info.local_size)
+            local_envs["HOROVOD_CROSS_RANK"] = str(alloc_info.cross_rank)
+            local_envs["HOROVOD_CROSS_SIZE"] = str(alloc_info.cross_size)
+
+    def run(self, func):
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/horovod/horovod_ray_trainer.py:208]: change to `train`

Comment (yangw1234) [reply pyzoo/zoo/orca/learn/horovod/horovod_ray_trainer.py]: changed

**Comment by yangw1234** [comment]
manually tested
other tests will be added in the following pr

**Pull request merged**